### PR TITLE
Change network input during training

### PR DESCRIPTION
**Pull request by guilhermebene**
Make network inputs for training and reconstruction follow the dimensions [b,c,h,w] according to #24.
Fix reconstruction with multiple channels described in #12.

### CODE REVIEW

**Review by nducros — COMMENTED** [review]
### Origin of the issue:
`compNet` and its children have several attributes of type `nn.Conv2d` that implement matrix-vector products. We found that `nn.Conv2d` is a faster alternative to `nn.Linear` for matrix-vector products. However, nn.Conv2d sums over the input number of channels while we would like to process them independently

### Solution:
* reshape the inputs of `nn.Conv2d` to merge batch and channel dimensions, back and forth

File: spyrit/learning/model_Had_DCAN.py [hunk 1]
```diff
@@ -700,24 +700,25 @@ def forward_reconstruct_mmse(self, x, b, c, h, w):
...
         x = x[:,:,self.even_index] - x[:,:,self.uneven_index];
-        x = 2*x-torch.reshape(self.Patt(torch.ones(b*c,1, h,w).to(x.device)),(b*c,1,self.M));
+        x = 2*x-torch.reshape(self.Patt(torch.ones(b*c,1,h,w).to(x.device)),(b,c,self.M));
         return x
     
     def forward_maptoimage(self, x, b, c, h, w):
         #--Projection to the image domain
         x = self.fc1(x);
-        x = x.view(b*c,1,h,w)
+        x = x.view(b, c, h, w)
```
Comment (nducros) [inline spyrit/learning/model_Had_DCAN.py:709]: Is this view necessary? x already (b, c, h, w) ?

File: spyrit/learning/model_Had_DCAN.py [hunk 2]
```diff
@@ -700,24 +700,25 @@ def forward_reconstruct_mmse(self, x, b, c, h, w):
...
     def forward_postprocess(self, x, b, c, h, w):
+        x = x.view(b*c, 1, h, w)
         x = self.recon(x)
         x = x.view(b, c, h, w)
         return x
     
     def pinv(self, x, b, c, h, w):
         x = self.Pinv(x);
-        x = x.view(b*c,1,h,w)
+        x = x.view(b, c, h, w)
```
Comment (nducros) [inline spyrit/learning/model_Had_DCAN.py:721]: Is `view` necessary ? `x` already `(b, c, h, w)` ?

File: spyrit/learning/model_Had_DCAN.py [hunk 3]
```diff
@@ -834,8 +835,8 @@ def forward(self, x):
         return x
     
     def forward_denoise(self, x, var, b, c, h, w):
```
Comment (nducros) [inline spyrit/learning/model_Had_DCAN.py:837]: Add `forward_denoise_expe(self, x, var, N0_est b, c, h, w)` where `N0_est` is `(b,c)`

File: spyrit/learning/model_Had_DCAN.py (hunk 1 above)
Comment (guilhermebene) [reply spyrit/learning/model_Had_DCAN.py:709]: Yes, because `self.fc1(x)` returns an array with dimensions [b,c,h*w].

**Pull request merged**